lua/ttt2/libraries/game_effects.lua

```diff
@@ -0,0 +1,124 @@
+---
+-- A library to consolidate some common effects code.
+-- @author EntranceJew
+-- @module gameEffects
+if SERVER then
+	AddCSLuaFile()
+end
+
+gameEffects = {}
+
+---
+-- Create a bundle of fires all from a central location.
+-- This is used for incendiary grenades or C4 detonation.
+-- @param Vector pos The position the fires should originate from.
+-- @param TraceResult tr A trace to orient the creation of the fires around.
+-- @param number num The number of individual balls of fire that should be created.
+-- @param number lifetime The base lifetime of all fires in the bundle.
+-- @param boolean explode Should the fires explode when they reach the end of their lives?
+-- @param nil|Player dmgowner The player to attribute the fire damage to.
+-- @param number spread_force The force that each fire will be flung with.
+-- @param boolean immobile If true, fires will become stationary once they begin burning.
+-- @param number size The physical scale of the fires.
+-- @param number lifetime_variance The amount each lifetime for each fire can vary.
+-- @return table A table full of the fire entities.
+-- @realm shared
+function gameEffects.StartFires(pos, tr, num, lifetime, explode, dmgowner, spread_force, immobile, size, lifetime_variance)
+	local flames = {}
+	for i = 1, num do
+		local ang = Angle(-math.Rand(0, 180), math.Rand(0, 360), math.Rand(0, 360))
+		local vstart = pos + tr.HitNormal * 64
+		local ttl = lifetime + math.Rand(-lifetime_variance, lifetime_variance)
+
+		if CreateVFireBall then
+			flames[#flames + 1] = CreateVFireBall(ttl, size, vstart, ang:Forward() * spread_force, dmgowner)
+			continue
+		end
```

Comment:
I'd remove this, because otherwise our extra handling in `SpawnFire` is never used and all the logic in ttt_flame is also not controlling this anymore.

Reply:
It still creates a managed instance of an entity. If we didn't create a fireball then it wouldn't spawn fire that can be merged with the vfire ball for anything that calls it directly.

Reply:
I don't understand what you are trying to explain sorry

Reply:
vfire when it exists is supposed to deal with everything pertaining to fire, all the appropriate parameters are remapped accordingly

Reply:
Then please add a note to the function doc block that if vFire is enabled, parameters such as immobile, explode, lifetime_variance, lifetime, num are ignored.
But then to be honest this function is not really doing what someone would expect, so my suggestion is to remove this if part here entirely and let this function take care of controlling how multiple fires are spawned while you can check for CreateVFireBall directly inside the firegrenade and call that instead of this function.

I think this would be the best solution.

Reply:
the functions aren't meant to just replace the grenade, they're meant to replace functions that other addons which create fire may have been using, like the C4 for example

Reply:
It can't, because no other addon uses this new gameeffects function. So i don't get your point here. It just does not make sense to have an entire SpawnFires function that just does not spawn fires but a single fireball.

Reply:
yeah, anything that did use them, should now use these

counter to its name, a fireball also explodes into several fires after impact which is why it's functionally similar

Reply:
Yeah no i still disagree here. Maybe @Histalek or @TimGoll can chime in